libs/sdk-ui-kit/src/RecurrenceForm/useCronValidation.ts

```diff
@@ -0,0 +1,146 @@
+// (C) 2024-2025 GoodData Corporation
+
+import { useCallback, useState } from "react";
+import cronParser from "cron-parser";
+import { isCronExpressionValid } from "./utils.js";
+import { defineMessages } from "react-intl";
+
+export enum RecurrenceFormCronErrorTypes {
+    EMPTY = "EMPTY",
+    WRONG_FORMAT = "WRONG_FORMAT",
+    FREQUENCY_TOO_HIGH = "FREQUENCY_TOO_HIGH",
+}
+
+const errorMessages = defineMessages({
+    empty: { id: "recurrence.error.empty" },
+    wrongFormat: { id: "recurrence.error.wrongFormat" },
+    frequencyTooHigh: { id: "recurrence.error.too_often" },
+});
+
+interface IUseCronValidationProps {
+    allowHourlyRecurrence: boolean;
+    onCronValueChange: (cronValue: string, isValid: boolean) => void;
+}
+
+export const useCronValidation = (props: IUseCronValidationProps) => {
+    const { allowHourlyRecurrence, onCronValueChange } = props;
+    const [cronError, setCronError] = useState<string | null>(null);
+
+    const parseCron = useCallback((expression: string) => {
+        try {
+            cronParser.CronExpressionParser.parse(expression);
+            return true;
+        } catch {
+            return false;
+        }
+    }, []);
+
+    const validate = useCallback(
+        (expression: string) => {
+            const trimmedCronExpression = expression.replace(/\s+/g, "");
+            const parts = expression.split(/\s+/).filter(Boolean);
+
+            if (trimmedCronExpression.length === 0) {
+                return RecurrenceFormCronErrorTypes.EMPTY;
+            }
+
+            const isValid = isCronExpressionValid(expression, allowHourlyRecurrence);
+            if (!isValid) {
+                return RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH;
+            }
+
+            if (parts.length !== 6) {
+                return RecurrenceFormCronErrorTypes.WRONG_FORMAT;
+            }
+
+            const isValidCron = parseCron(expression);
```

Comment:
I would switch order of checks:
1 - validity of cron expression by parseCron, does it cover also too short expression? If yes then check for parts length is redundant
2 - frequency check - it does not make sense to check frequency, if expression itself is not valid

Reply:
Switched order.
Regarding the parts check, cron-parser treats now seconds as optional, still decided to use this package since repo looks active and is widely used. Have to keep this check of parts.

Reply:
ok